api/api.go

```diff
@@ -1117,7 +1139,7 @@ func generateUnexpectedResponseCodeError(resp *http.Response) error {
 	var buf bytes.Buffer
 	io.Copy(&buf, resp.Body)
 	closeResponseBody(resp)
-	return fmt.Errorf("Unexpected response code: %d (%s)", resp.StatusCode, buf.Bytes())
+	return StatusError{Code: resp.StatusCode, err: errors.New(string(buf.Bytes()))}
```

Comment:
I don't see too much value in keeping `err error` field since we convert from []byte -> string -> error -> string (when output as `%v` from `Error()`)
Would it make more sense as `body string`?

Reply:
Thanks for pointing that out -- agreed 💯 ✅